algorithms/unit_tests/TestStdAlgorithmsTeamAdjacentDifference.cpp

```diff
@@ -0,0 +1,206 @@
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <TestStdAlgorithmsCommon.hpp>
+#include <cstddef>
+#include <functional>
+#include <numeric>
+#include "gtest/gtest.h"
+#include "std_algorithms/Kokkos_BeginEnd.hpp"
+
+namespace Test {
+namespace stdalgos {
+namespace TestAdjacentDifference {
+
+namespace KE = Kokkos::Experimental;
+
+template <class ValueType>
+struct PlusFunctor {
+  KOKKOS_INLINE_FUNCTION constexpr ValueType operator()(
+      const ValueType& lhs, const ValueType& rhs) const {
+    return lhs + rhs;
+  }
+};
+
+template <class ViewType, class DiffsViewType, class BinaryOp>
+struct TestFunctorA {
+  ViewType m_view;
+  DiffsViewType m_diffsView;
+  int m_apiPick;
+  BinaryOp m_binaryOp;
+
+  TestFunctorA(const ViewType view, const DiffsViewType diffsView, int apiPick,
+               BinaryOp binaryOp)
+      : m_view(view),
+        m_diffsView(diffsView),
+        m_apiPick(apiPick),
+        m_binaryOp(std::move(binaryOp)) {}
+
+  template <class MemberType>
+  KOKKOS_INLINE_FUNCTION void operator()(const MemberType& member) const {
+    const auto myRowIndex = member.league_rank();
+
+    auto myRowView  = Kokkos::subview(m_view, myRowIndex, Kokkos::ALL());
+    auto myDiffView = Kokkos::subview(m_diffsView, myRowIndex, Kokkos::ALL());
+
+    switch (m_apiPick) {
+      case 0: {
+        KE::adjacent_difference(member, KE::begin(myRowView),
+                                KE::end(myRowView), KE::begin(myDiffView));
+        break;
+      }
+
+      case 1: {
+        KE::adjacent_difference(member, KE::begin(myRowView),
+                                KE::end(myRowView), KE::begin(myDiffView),
+                                m_binaryOp);
+        break;
+      }
+
+      case 2: {
+        KE::adjacent_difference(member, myRowView, myDiffView);
+        break;
+      }
+
+      case 3: {
+        KE::adjacent_difference(member, myRowView, myDiffView, m_binaryOp);
+        break;
+      }
+    }
+  }
+};
+
+template <class LayoutTag, class ValueType>
+void test_A(std::size_t numTeams, std::size_t numCols, int apiId) {
+  /* description:
+     use a rank-2 view randomly filled with values,
+     and run a team-level adjacent_difference
+   */
+
+  // -----------------------------------------------
+  // prepare data
+  // -----------------------------------------------
+  // create a view in the memory space associated with default exespace
+  // with as many rows as the number of teams and fill it with random
+  // values from an arbitrary range.
+  auto [dataView, cloneOfDataViewBeforeOp_h] =
```

Comment:
call it sourceView (see copy_if for example)

Reply:
Done.